sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/BaseClientBuilder.java

```diff
@@ -98,6 +102,29 @@ protected HttpPipeline buildPipeline() {
             .build();
     }
 
+    /**
+     * Creates a policy that makes assertions on HTTP responses. These assertions are general purpose; method-specific
+     * validations should be performed in the convenience layer.
+     *
+     * @return The validation policy.
+     */
+    private HttpPipelinePolicy makeValidationPolicy() {
+        ResponseValidationPolicyBuilder builder = new ResponseValidationPolicyBuilder()
+            .optionalEcho(Constants.HeaderConstants.CLIENT_REQUEST_ID); // echo client request id
+
+        applyServiceSpecificValidations(builder);
+
+        return builder.build();
+    }
+
+    /**
+     * Applies validation of general-purpose requests to builder. Method-specific validations should be performed in the
+     * convenience layer.
+     *
+     * @param builder Builder to assemble assertions together.
+     */
+    protected abstract void applyServiceSpecificValidations(ResponseValidationPolicyBuilder builder);
```

Comment:
I did not see any implementation on each builder. Is this method a placeholder or leave it here for some other purpose?

Reply:
This this is overridden with `final` in BaseBlobClientBuilder, BaseFileClientBuilder, and BaseQueueClientBuilder. They are currently all three no-ops, but this is laying the groundwork for CPK to validate that the key-hash returned is the same as the one sent over the wire. CPK is a blob-only feature, so there was no need to add extra steps to file and queue's pipelines.